core/field_textinput.js

```diff
@@ -432,4 +432,16 @@ Blockly.FieldTextInput.nonnegativeIntegerValidator = function(text) {
   return n;
 };
 
+/**
+ * Get the size of the visible field, as used in new rendering.
+ * @return {!goog.math.Size} The size of the visible field.
+ * @package
+ */
+Blockly.FieldTextInput.prototype.getCorrectedSize = function() {
+  // getSize also renders and updates the size if needed.  Rather than duplicate
+  // the logic to figure out whether to rerender, just call getSize.
+  this.getSize();
+  return new goog.math.Size(this.size_.width + Blockly.BlockSvg.SEP_SPACE_X, 16);
```

Comment:
I'm surprised the height isn't dependent on the text. What happens if you change the css for fonts?

Reply:
The height doesn't change.  This is partly because SVG has `getComputedTextWidth` but no matching `getComputedTextHeight`